rasa/core/policies/policy.py

```diff
@@ -285,6 +293,15 @@ def load(cls, path: Union[Text, Path]) -> "Policy":
 
         if metadata_file.is_file():
             data = json.loads(rasa.shared.utils.io.read_file(metadata_file))
+            if should_finetune:
+                if "should_finetune" not in rasa.shared.utils.common.arguments_of(cls):
+                    raise UnsupportedDialogueModelError(
+                        f"{cls.__name__} does not support fine-tuning. "
+                        f"To support fine-tuning the `__init__` method must have the "
+                        f"argument `should_finetune`. This argument should be added to "
+                        f"all policies by Rasa Open Source 3.0.0."
+                    )
+                data["should_finetune"] = should_finetune
```

Comment:
I am not sure if adding this to the base class makes a lot of sense. Ability to finetune a model depends on the type of the model. While deep learning based models need a specific implementation to finetune themselves, other simple ML models do not and it's easier to just "train" them on the complete new data. So, I would not make `should_finetune` a parameter of the constructor of `Policy`. It depends on the model of the `Policy` if it needs to be there.
The idea is similar to what we do for NLU components. We haven't made `should_finetune` a parameter of the `Component` class but a parameter for specific ML components which require it.

Reply:
If we remove this argument from the constructor, then custom policies which do not differ in implementation for training the policy from scratch and training the policy in finetune mode will not have to be refactored at all.

Reply:
I see your argument. What about having a `context` dictionary as argument for the constructor for **all** policies? That would also solve the `epoch_override` problem neatly. Using a `context` argument we'd have 
1) A common interface for all policies
2) Policies can decide themselves if they want to handle it
3) Similar approach to NLU which makes the codebase quicker to understand

Reply:
Yes, having a context dictionary sounds like a good idea 👍

Reply:
There are two interfaces to a `Policy`, `load` and `__init__`.
`load` is called from the ensemble, and currently we are supporting:
```
def load( cls, path: Union[Text, Path], should_finetune: bool = False, epoch_override: Optional[float] = None)
def load( cls, path: Union[Text, Path], should_finetune: bool = False)
def load( cls, path: Union[Text, Path])
```
The next interface, `__init__` is called from the base class `Policy.load`, and currently we are supporting:
```
def __init__(self, args, should_finetune: bool = False)
def __init__(self, args)
```
Policies like `TEDPolicy` have a custom load, so they then handle passing `epochs` into their constructor.

So, should we have a `context` (essentially like the `kwargs` in my first draft version) in both?
And do we remove all warnings related to arguments? Or only warn if `context` is not there?